src/Build.UnitTests/Graph/ProjectGraph_Tests.cs

```diff
@@ -37,6 +46,220 @@ public void TestGraphWithSingleNode()
                 projectNode.FullPath.ShouldBe(testProject.ProjectFile);
             }
         }
+
+        /// <summary>
+        ///  A  
+        /// / \ 
+        ///B   C
+        /// </summary>
+        [Fact]
+        public void TestGraphWithThreeNodes()
+        {
+            string projectAContents = @"
+                <Project xmlns='http://schemas.microsoft.com/developer/msbuild/2003'>
+                  <PropertyGroup>
+                    <TestProperty>value</TestProperty>
+                  </PropertyGroup>
+                  <ItemGroup>
+                    <ProjectReference Include=""B.proj"" />
+                    <ProjectReference Include=""C.proj"" />
+                  </ItemGroup>
+                </Project>
+                ";
+
+            string projectBContents = @"
+                <Project xmlns='http://schemas.microsoft.com/developer/msbuild/2003'>
+                  <PropertyGroup>
+                    <TestProperty>value</TestProperty>
+                  </PropertyGroup>
+                </Project>
+                ";
+
+            string projectCContents = @"
+                <Project xmlns='http://schemas.microsoft.com/developer/msbuild/2003'>
+                  <PropertyGroup>
+                    <TestProperty>value</TestProperty>
+                  </PropertyGroup>
+                </Project>
+                ";
+
+            using (var env = TestEnvironment.Create())
+            {
+                TransientTestProjectWithFiles projectA =
+                    env.CreateTestProjectWithFiles(ProjA, projectAContents, new []{ProjB, ProjC});
+                File.WriteAllText(Path.Combine(projectA.TestRoot, ProjB), projectBContents);
+                File.WriteAllText(Path.Combine(projectA.TestRoot, ProjC), projectCContents);
+                ProjectGraph graph = new ProjectGraph(projectA.ProjectFile);
+
+                graph.ProjectNodes.Count.ShouldBe(3);
+                ProjectGraphNode projectNodeA = graph.ProjectNodes.First(node => node.Project.FullPath.Equals(projectA.ProjectFile));
+                ProjectGraphNode projectNodeB = graph.ProjectNodes.First(node => node.Project.FullPath.Contains(ProjB));
+                ProjectGraphNode projectNodeC = graph.ProjectNodes.First(node => node.Project.FullPath.Contains(ProjC));
+                projectNodeA.ProjectReferences.Count.ShouldBe(2);
+                projectNodeB.ProjectReferences.Count.ShouldBe(0);
+                projectNodeC.ProjectReferences.Count.ShouldBe(0);
+            }
+        }
+
+        /// <summary>
+        /// Test the following graph with entry project B
```

Comment:
In case it helps anyone else:

![image](https://user-images.githubusercontent.com/3347530/45723870-4ac1ae00-bb68-11e8-911b-3d73e9f4c741.png)